src/App.tsx

```diff
@@ -4,82 +4,63 @@ import { useDispatch, useSelector } from "react-redux";
 import { GetUser } from "./actions/UserActions";
 import { UserType } from "./actions/UserActionTypes";
 import UserPagination from "./components/UserPagination/UserPagination";
-import { RootStore } from "./store";
+import { UserStore } from "./store";
+import Heading from "./components/Heading/Heading";
+import UserList from "./components/UserList/UserList";
+import UserProfile from "./components/UserProfile/Userprofile";
 
 export interface DataType {
   selected: number;
 }
 
 function App() {
   const dispatch = useDispatch();
-  const userState = useSelector((state: RootStore) => state.users);
-  const [userdata, setUserData] = useState<UserType[] | [] | undefined>([]);
+  const userState = useSelector((state: UserStore) => state);
+  const [selectedUser, setSelectedUser] = useState<UserType | null>(null);
   const [pagenum, setPageNum] = useState(1);
-  console.log("user state:", userState);
   const handlePageClick = (data: DataType) => {
     setPageNum(data.selected + 1);
   };
 
-  useEffect(() => {
-    console.log("number is", pagenum);
-    dispatch(GetUser(pagenum));
+  const handleHover = (id: number) => {
     if (userState.users) {
-      setUserData(userState.users);
+      const selectedUserInfo = userState.users.find((user) => user.id === id);
+      if (selectedUserInfo) {
+        setSelectedUser(selectedUserInfo);
+      }
     }
-  }, [pagenum]);
+  };
 
+  const handleMouseLeave = () => {
+    setSelectedUser(null);
+  };
+
+  const profile_info = selectedUser !== null && (
+    <UserProfile selectedUser={selectedUser} />
+  );
   useEffect(() => {
-    setUserData(userState.users);
-  }, []);
+    dispatch(GetUser(pagenum));
+  }, [pagenum, dispatch]);
```

Comment:
This side effect to fetch data is duplicated in UserList component. 
As a result, will call the API 2 times. Call it once and access it everywhere since it is stored in redux.